flake: TestIntegrationTestSuite/TestMaxBlockSize

Observed in merge-queue CI run for #7079:

https://github.com/celestiaorg/celestia-app/actions/runs/24479285808/job/71539583567

Job: `test / go-test (github.com/celestiaorg/celestia-app/v9/app/test, TestIntegration.*)`

```
--- FAIL: TestIntegrationTestSuite (60.84s)
    --- FAIL: TestIntegrationTestSuite/TestMaxBlockSize (33.97s)
        integration_test.go:101:
            Error Trace:	/home/runner/work/celestia-app/celestia-app/app/test/integration_test.go:101
                        	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.11.1/suite/suite.go:115
            Error:      	Received unexpected error:
                        	timeout (30s) exceeded waiting for network to reach height 20. Got to height 12
            Test:       	TestIntegrationTestSuite/TestMaxBlockSize
        --- FAIL: TestIntegrationTestSuite/TestMaxBlockSize/randomTxGen (33.97s)
            testing.go:1913: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
    network.go:79: tearing down testnode
    network.go:120: error when closing API server close tcp 127.0.0.1:33188: use of closed network connection
FAIL	github.com/celestiaorg/celestia-app/v9/app/test	60.917s
```

### Context

- PR #7079 does not touch `app/test/integration_test.go` or any code in the consensus/block-production path — it only adds a 6-line skip in `test/docker-e2e/e2e_full_stack_pfb_test.go`. The failure is unrelated to the PR.
- The test broadcasts 60 blob txs of ~1 MiB each (`integration_test.go:66-77`), then calls `s.cctx.WaitForBlocks(10)` at line 101, which uses the 30 s `DefaultTimeout` in `test/util/testnode/node_interaction_api.go:30`.
- Consensus was still making progress on the runner (height advanced 10 → 12 during the 30 s window) — it was just slow, consistent with a loaded GitHub runner struggling to produce 10 blocks while processing 60 × ~1 MiB blob txs.
- Sampling the last ~10 merge-queue failures, this job has not reappeared, so the flake rate appears low. Filing this to have a tracker the next time it shows up.

### Prior flake history

- #574 TestIntegrationTestSuite/TestMaxBlockSize flake
- #3086 TestMaxBlockSize flake
- #3089, #3150, #3154, #3183 — previous fix attempts

### Next steps

- Try reproducing locally via stress-run (`go test -count=N -run TestIntegrationTestSuite/TestMaxBlockSize …`).
- If reproducible, consider bumping the `WaitForBlocks(10)` timeout for this specific call, or reducing the tx volume to make the test less sensitive to runner load.

## Reproduced locally under CPU contention

### Baseline (idle machine)
10/10 passes, ~25 s each:
```
$ go test -count=10 -timeout=30m -run 'TestIntegrationTestSuite/TestMaxBlockSize$' github.com/celestiaorg/celestia-app/v9/app/test
ok  	github.com/celestiaorg/celestia-app/v9/app/test	249.068s
```

### Under CPU contention (6 copies in parallel × 3 iterations each)
**6/6 workers failed** with the exact CI error shape:
```
--- FAIL: TestIntegrationTestSuite/TestMaxBlockSize (32.04s)
    Error: timeout (30s) exceeded waiting for network to reach height 17. Got to height 8
--- FAIL: TestIntegrationTestSuite/TestMaxBlockSize (32.41s)
    Error: timeout (30s) exceeded waiting for network to reach height 18. Got to height 8
--- FAIL: TestIntegrationTestSuite/TestMaxBlockSize (32.38s)
    Error: timeout (30s) exceeded waiting for network to reach height 18. Got to height 9
```
(Host: M-series Mac, macOS 15.6, go1.26.1 darwin/arm64)

### Root cause
The test broadcasts 60 × ~1 MiB blob txs, then calls `s.cctx.WaitForBlocks(10)` at [`integration_test.go:101`](https://github.com/celestiaorg/celestia-app/blob/main/app/test/integration_test.go#L101), which bakes in the 30 s `DefaultTimeout` from [`test/util/testnode/node_interaction_api.go:30`](https://github.com/celestiaorg/celestia-app/blob/main/test/util/testnode/node_interaction_api.go#L30).

Needed throughput: **10 blocks in 30 s ≈ 0.33 blocks/s**. Under CPU pressure, each block has to ingest/sign/DA-encode up to ~1 MiB of blob data plus honor the 500 ms precommit timeout (`integration_test.go:48`). That's plausibly >3 s/block on a loaded 4-vCPU GitHub runner, which blows the 30 s budget.

Consensus was *still making progress* during the failure (height advanced); it's purely a timing-budget miss, not a liveness bug.

### Proposed fix
Replace the implicit 30 s timeout at `integration_test.go:101` with an explicit, generous one via `WaitForHeightWithTimeout`, e.g. 2 min. This keeps correctness assertions intact and just accommodates loaded CI runners. Will put up a PR.

--- a/app/test/integration_test.go
+++ b/app/test/integration_test.go
@@ -98,7 +98,13 @@ func (s *IntegrationTestSuite) TestMaxBlockSize() {
 				hashes[i] = res.TxHash
 			}
 
-			require.NoError(t, s.cctx.WaitForBlocks(10))
+			// Use a generous timeout because producing 10 blocks that each
+			// contain up to ~1 MiB of blob txs is CPU-heavy and the default
+			// 30s timeout flakes on loaded CI runners. See #7080.
+			lastHeight, err := s.cctx.LatestHeight()
+			require.NoError(t, err)
+			_, err = s.cctx.WaitForHeightWithTimeout(lastHeight+10, 2*time.Minute)
+			require.NoError(t, err)
 
 			heights := make(map[int64]int)
 			for _, hash := range hashes {

